In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns


In [2]:
df = pd.read_csv("census-income .csv")

In [3]:
df

age         workclass  fnlwgt   education  education-num  \
0       39         State-gov   77516   Bachelors             13   
1       50  Self-emp-not-inc   83311   Bachelors             13   
2       38           Private  215646     HS-grad              9   
3       53           Private  234721        11th              7   
4       28           Private  338409   Bachelors             13   
...    ...               ...     ...         ...            ...   
32556   27           Private  257302  Assoc-acdm             12   
32557   40           Private  154374     HS-grad              9   
32558   58           Private  151910     HS-grad              9   
32559   22           Private  201490     HS-grad              9   
32560   52      Self-emp-inc  287927     HS-grad              9   

           marital-status         occupation   relationship   race     sex  \
0           Never-married       Adm-clerical  Not-in-family  White    Male   
1      Married-civ-spouse    Exec-managerial        Husband  White    Male   
2                Divorced  Handlers-cleaners  Not-in-family  White    Male   
3      Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4      Married-civ-spouse     Prof-specialty           Wife  Black  Female   
...                   ...                ...            ...    ...     ...   
32556  Married-civ-spouse       Tech-support           Wife  White  Female   
32557  Married-civ-spouse  Machine-op-inspct        Husband  White    Male   
32558             Widowed       Adm-clerical      Unmarried  White  Female   
32559       Never-married       Adm-clerical      Own-child  White    Male   
32560  Married-civ-spouse    Exec-managerial           Wife  White  Female   

       capital-gain  capital-loss  hours-per-week native-country annual_income  
0              2174             0              40  United-States         <=50K  
1                 0             0              13  United-States         <=50K  
2                 0             0              40  United-States         <=50K  
3                 0             0              40  United-States         <=50K  
4                 0             0              40           Cuba         <=50K  
...             ...           ...             ...            ...           ...  
32556             0             0              38  United-States         <=50K  
32557             0             0              40  United-States          >50K  
32558             0             0              40  United-States         <=50K  
32559             0             0              20  United-States         <=50K  
32560         15024             0              40  United-States          >50K  

[32561 rows x 15 columns]

In [4]:
df.columns

Index(['age', 'workclass', 'fnlwgt', 'education', 'education-num',
       'marital-status', 'occupation', 'relationship', 'race', 'sex',
       'capital-gain', 'capital-loss', 'hours-per-week', 'native-country',
       'annual_income'],
      dtype='object')

In [5]:
df["capital_change"]=df["capital-gain"]-df["capital-loss"]

In [6]:
df["capital_change"]

0         2174
1            0
2            0
3            0
4            0
         ...  
32556        0
32557        0
32558        0
32559        0
32560    15024
Name: capital_change, Length: 32561, dtype: int64

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             32561 non-null  int64 
 1   workclass       32561 non-null  object
 2   fnlwgt          32561 non-null  int64 
 3   education       32561 non-null  object
 4   education-num   32561 non-null  int64 
 5   marital-status  32561 non-null  object
 6   occupation      32561 non-null  object
 7   relationship    32561 non-null  object
 8   race            32561 non-null  object
 9   sex             32561 non-null  object
 10  capital-gain    32561 non-null  int64 
 11  capital-loss    32561 non-null  int64 
 12  hours-per-week  32561 non-null  int64 
 13  native-country  32561 non-null  object
 14  annual_income   32561 non-null  object
 15  capital_change  32561 non-null  int64 
dtypes: int64(7), object(9)
memory usage: 4.0+ MB


In [8]:
df.isnull().sum()

age               0
workclass         0
fnlwgt            0
education         0
education-num     0
marital-status    0
occupation        0
relationship      0
race              0
sex               0
capital-gain      0
capital-loss      0
hours-per-week    0
native-country    0
annual_income     0
capital_change    0
dtype: int64

In [9]:
df['workclass'].value_counts()

Private             22696
Self-emp-not-inc     2541
Local-gov            2093
?                    1836
State-gov            1298
Self-emp-inc         1116
Federal-gov           960
Without-pay            14
Never-worked            7
Name: workclass, dtype: int64

In [10]:
df.loc[df['workclass']=='?']

age workclass  fnlwgt     education  education-num  \
27      54         ?  180211  Some-college             10   
61      32         ?  293936       7th-8th              4   
69      25         ?  200681  Some-college             10   
77      67         ?  212759          10th              6   
106     17         ?  304873          10th              6   
...    ...       ...     ...           ...            ...   
32530   35         ?  320084     Bachelors             13   
32531   30         ?   33811     Bachelors             13   
32539   71         ?  287372     Doctorate             16   
32541   41         ?  202822       HS-grad              9   
32542   72         ?  129912       HS-grad              9   

              marital-status occupation   relationship                race  \
27        Married-civ-spouse          ?        Husband  Asian-Pac-Islander   
61     Married-spouse-absent          ?  Not-in-family               White   
69             Never-married          ?      Own-child               White   
77        Married-civ-spouse          ?        Husband               White   
106            Never-married          ?      Own-child               White   
...                      ...        ...            ...                 ...   
32530     Married-civ-spouse          ?           Wife               White   
32531          Never-married          ?  Not-in-family  Asian-Pac-Islander   
32539     Married-civ-spouse          ?        Husband               White   
32541              Separated          ?  Not-in-family               Black   
32542     Married-civ-spouse          ?        Husband               White   

          sex  capital-gain  capital-loss  hours-per-week native-country  \
27       Male             0             0              60          South   
61       Male             0             0              40              ?   
69       Male             0             0              40  United-States   
77       Male             0             0               2  United-States   
106    Female         34095             0              32  United-States   
...       ...           ...           ...             ...            ...   
32530  Female             0             0              55  United-States   
32531  Female             0             0              99  United-States   
32539    Male             0             0              10  United-States   
32541  Female             0             0              32  United-States   
32542    Male             0             0              25  United-States   

      annual_income  capital_change  
27             >50K               0  
61            <=50K               0  
69            <=50K               0  
77            <=50K               0  
106           <=50K           34095  
...             ...             ...  
32530          >50K               0  
32531         <=50K               0  
32539          >50K               0  
32541         <=50K               0  
32542         <=50K               0  

[1836 rows x 16 columns]

In [11]:
df.loc[df['occupation']=='?']

age workclass  fnlwgt     education  education-num  \
27      54         ?  180211  Some-college             10   
61      32         ?  293936       7th-8th              4   
69      25         ?  200681  Some-college             10   
77      67         ?  212759          10th              6   
106     17         ?  304873          10th              6   
...    ...       ...     ...           ...            ...   
32530   35         ?  320084     Bachelors             13   
32531   30         ?   33811     Bachelors             13   
32539   71         ?  287372     Doctorate             16   
32541   41         ?  202822       HS-grad              9   
32542   72         ?  129912       HS-grad              9   

              marital-status occupation   relationship                race  \
27        Married-civ-spouse          ?        Husband  Asian-Pac-Islander   
61     Married-spouse-absent          ?  Not-in-family               White   
69             Never-married          ?      Own-child               White   
77        Married-civ-spouse          ?        Husband               White   
106            Never-married          ?      Own-child               White   
...                      ...        ...            ...                 ...   
32530     Married-civ-spouse          ?           Wife               White   
32531          Never-married          ?  Not-in-family  Asian-Pac-Islander   
32539     Married-civ-spouse          ?        Husband               White   
32541              Separated          ?  Not-in-family               Black   
32542     Married-civ-spouse          ?        Husband               White   

          sex  capital-gain  capital-loss  hours-per-week native-country  \
27       Male             0             0              60          South   
61       Male             0             0              40              ?   
69       Male             0             0              40  United-States   
77       Male             0             0               2  United-States   
106    Female         34095             0              32  United-States   
...       ...           ...           ...             ...            ...   
32530  Female             0             0              55  United-States   
32531  Female             0             0              99  United-States   
32539    Male             0             0              10  United-States   
32541  Female             0             0              32  United-States   
32542    Male             0             0              25  United-States   

      annual_income  capital_change  
27             >50K               0  
61            <=50K               0  
69            <=50K               0  
77            <=50K               0  
106           <=50K           34095  
...             ...             ...  
32530          >50K               0  
32531         <=50K               0  
32539          >50K               0  
32541         <=50K               0  
32542         <=50K               0  

[1843 rows x 16 columns]

In [12]:
df.loc[df['native-country']=='?']

age         workclass  fnlwgt     education  education-num  \
14      40           Private  121772     Assoc-voc             11   
38      31           Private   84154  Some-college             10   
51      18           Private  226956       HS-grad              9   
61      32                 ?  293936       7th-8th              4   
93      30           Private  117747       HS-grad              9   
...    ...               ...     ...           ...            ...   
32449   44      Self-emp-inc   71556       Masters             14   
32469   58      Self-emp-inc  181974     Doctorate             16   
32492   42  Self-emp-not-inc  217597       HS-grad              9   
32510   39           Private  107302       HS-grad              9   
32525   81                 ?  120478     Assoc-voc             11   

              marital-status      occupation   relationship  \
14        Married-civ-spouse    Craft-repair        Husband   
38        Married-civ-spouse           Sales        Husband   
51             Never-married   Other-service      Own-child   
61     Married-spouse-absent               ?  Not-in-family   
93        Married-civ-spouse           Sales           Wife   
...                      ...             ...            ...   
32449     Married-civ-spouse           Sales        Husband   
32469          Never-married  Prof-specialty  Not-in-family   
32492               Divorced           Sales      Own-child   
32510     Married-civ-spouse  Prof-specialty        Husband   
32525               Divorced               ?      Unmarried   

                     race     sex  capital-gain  capital-loss  hours-per-week  \
14     Asian-Pac-Islander    Male             0             0              40   
38                  White    Male             0             0              38   
51                  White  Female             0             0              30   
61                  White    Male             0             0              40   
93     Asian-Pac-Islander  Female             0          1573              35   
...                   ...     ...           ...           ...             ...   
32449               White    Male             0             0              50   
32469               White  Female             0             0              99   
32492               White    Male             0             0              50   
32510               White    Male             0             0              45   
32525               White  Female             0             0               1   

      native-country annual_income  capital_change  
14                 ?          >50K               0  
38                 ?          >50K               0  
51                 ?         <=50K               0  
61                 ?         <=50K               0  
93                 ?         <=50K           -1573  
...              ...           ...             ...  
32449              ?          >50K               0  
32469              ?         <=50K               0  
32492              ?         <=50K               0  
32510              ?          >50K               0  
32525              ?         <=50K               0  

[583 rows x 16 columns]

In [13]:
df[(df['workclass']=='Private') & (df['native-country']=='United-States')]

age workclass  fnlwgt     education  education-num      marital-status  \
2       38   Private  215646       HS-grad              9            Divorced   
3       53   Private  234721          11th              7  Married-civ-spouse   
5       37   Private  284582       Masters             14  Married-civ-spouse   
8       31   Private   45781       Masters             14       Never-married   
9       42   Private  159449     Bachelors             13  Married-civ-spouse   
...    ...       ...     ...           ...            ...                 ...   
32555   22   Private  310152  Some-college             10       Never-married   
32556   27   Private  257302    Assoc-acdm             12  Married-civ-spouse   
32557   40   Private  154374       HS-grad              9  Married-civ-spouse   
32558   58   Private  151910       HS-grad              9             Widowed   
32559   22   Private  201490       HS-grad              9       Never-married   

              occupation   relationship   race     sex  capital-gain  \
2      Handlers-cleaners  Not-in-family  White    Male             0   
3      Handlers-cleaners        Husband  Black    Male             0   
5        Exec-managerial           Wife  White  Female             0   
8         Prof-specialty  Not-in-family  White  Female         14084   
9        Exec-managerial        Husband  White    Male          5178   
...                  ...            ...    ...     ...           ...   
32555    Protective-serv  Not-in-family  White    Male             0   
32556       Tech-support           Wife  White  Female             0   
32557  Machine-op-inspct        Husband  White    Male             0   
32558       Adm-clerical      Unmarried  White  Female             0   
32559       Adm-clerical      Own-child  White    Male             0   

       capital-loss  hours-per-week native-country annual_income  \
2                 0              40  United-States         <=50K   
3                 0              40  United-States         <=50K   
5                 0              40  United-States         <=50K   
8                 0              50  United-States          >50K   
9                 0              40  United-States          >50K   
...             ...             ...            ...           ...   
32555             0              40  United-States         <=50K   
32556             0              38  United-States         <=50K   
32557             0              40  United-States          >50K   
32558             0              40  United-States         <=50K   
32559             0              20  United-States         <=50K   

       capital_change  
2                   0  
3                   0  
5                   0  
8               14084  
9                5178  
...               ...  
32555               0  
32556               0  
32557               0  
32558               0  
32559               0  

[20135 rows x 16 columns]

In [14]:
df.drop(df[df['workclass']=='?'].index, inplace = True)

In [15]:
df.drop(df[df['occupation']=='?'].index, inplace = True)

In [16]:
df

age         workclass  fnlwgt   education  education-num  \
0       39         State-gov   77516   Bachelors             13   
1       50  Self-emp-not-inc   83311   Bachelors             13   
2       38           Private  215646     HS-grad              9   
3       53           Private  234721        11th              7   
4       28           Private  338409   Bachelors             13   
...    ...               ...     ...         ...            ...   
32556   27           Private  257302  Assoc-acdm             12   
32557   40           Private  154374     HS-grad              9   
32558   58           Private  151910     HS-grad              9   
32559   22           Private  201490     HS-grad              9   
32560   52      Self-emp-inc  287927     HS-grad              9   

           marital-status         occupation   relationship   race     sex  \
0           Never-married       Adm-clerical  Not-in-family  White    Male   
1      Married-civ-spouse    Exec-managerial        Husband  White    Male   
2                Divorced  Handlers-cleaners  Not-in-family  White    Male   
3      Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4      Married-civ-spouse     Prof-specialty           Wife  Black  Female   
...                   ...                ...            ...    ...     ...   
32556  Married-civ-spouse       Tech-support           Wife  White  Female   
32557  Married-civ-spouse  Machine-op-inspct        Husband  White    Male   
32558             Widowed       Adm-clerical      Unmarried  White  Female   
32559       Never-married       Adm-clerical      Own-child  White    Male   
32560  Married-civ-spouse    Exec-managerial           Wife  White  Female   

       capital-gain  capital-loss  hours-per-week native-country  \
0              2174             0              40  United-States   
1                 0             0              13  United-States   
2                 0             0              40  United-States   
3                 0             0              40  United-States   
4                 0             0              40           Cuba   
...             ...           ...             ...            ...   
32556             0             0              38  United-States   
32557             0             0              40  United-States   
32558             0             0              40  United-States   
32559             0             0              20  United-States   
32560         15024             0              40  United-States   

      annual_income  capital_change  
0             <=50K            2174  
1             <=50K               0  
2             <=50K               0  
3             <=50K               0  
4             <=50K               0  
...             ...             ...  
32556         <=50K               0  
32557          >50K               0  
32558         <=50K               0  
32559         <=50K               0  
32560          >50K           15024  

[30718 rows x 16 columns]

In [17]:
df.dtypes

age                int64
workclass         object
fnlwgt             int64
education         object
education-num      int64
marital-status    object
occupation        object
relationship      object
race              object
sex               object
capital-gain       int64
capital-loss       int64
hours-per-week     int64
native-country    object
annual_income     object
capital_change     int64
dtype: object

In [18]:
df.shape

(30718, 16)

In [19]:
df.duplicated().sum()

24

In [20]:
df[(df['occupation']=='Tech-support') & (df['annual_income']=='>50K')].shape

(283, 16)

In [21]:
df=df.drop_duplicates()
df


age         workclass  fnlwgt   education  education-num  \
0       39         State-gov   77516   Bachelors             13   
1       50  Self-emp-not-inc   83311   Bachelors             13   
2       38           Private  215646     HS-grad              9   
3       53           Private  234721        11th              7   
4       28           Private  338409   Bachelors             13   
...    ...               ...     ...         ...            ...   
32556   27           Private  257302  Assoc-acdm             12   
32557   40           Private  154374     HS-grad              9   
32558   58           Private  151910     HS-grad              9   
32559   22           Private  201490     HS-grad              9   
32560   52      Self-emp-inc  287927     HS-grad              9   

           marital-status         occupation   relationship   race     sex  \
0           Never-married       Adm-clerical  Not-in-family  White    Male   
1      Married-civ-spouse    Exec-managerial        Husband  White    Male   
2                Divorced  Handlers-cleaners  Not-in-family  White    Male   
3      Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4      Married-civ-spouse     Prof-specialty           Wife  Black  Female   
...                   ...                ...            ...    ...     ...   
32556  Married-civ-spouse       Tech-support           Wife  White  Female   
32557  Married-civ-spouse  Machine-op-inspct        Husband  White    Male   
32558             Widowed       Adm-clerical      Unmarried  White  Female   
32559       Never-married       Adm-clerical      Own-child  White    Male   
32560  Married-civ-spouse    Exec-managerial           Wife  White  Female   

       capital-gain  capital-loss  hours-per-week native-country  \
0              2174             0              40  United-States   
1                 0             0              13  United-States   
2                 0             0              40  United-States   
3                 0             0              40  United-States   
4                 0             0              40           Cuba   
...             ...           ...             ...            ...   
32556             0             0              38  United-States   
32557             0             0              40  United-States   
32558             0             0              40  United-States   
32559             0             0              20  United-States   
32560         15024             0              40  United-States   

      annual_income  capital_change  
0             <=50K            2174  
1             <=50K               0  
2             <=50K               0  
3             <=50K               0  
4             <=50K               0  
...             ...             ...  
32556         <=50K               0  
32557          >50K               0  
32558         <=50K               0  
32559         <=50K               0  
32560          >50K           15024  

[30694 rows x 16 columns]

In [22]:
df[(df['native-country']!='United-States') & (df['workclass']=='Private')].shape

(2554, 16)

In [23]:
#How many people are either having Annual Income(last column) less than or
#equal to 50k or their working hours is greater than or equal to 40 hrs:
df[(df['annual_income']=='<=50K') & (df['hours-per-week']>=40)].shape

(16901, 16)

In [24]:
df['occupation'].nunique()

14

In [25]:
# For checking the employees in Tech support team earning more than 50k
df[(df['occupation']=='Tech-support') & (df['annual_income']=='>50K')].shape

(283, 16)

In [26]:
df['annual_income'] == '>50k'

0        False
1        False
2        False
3        False
4        False
         ...  
32556    False
32557    False
32558    False
32559    False
32560    False
Name: annual_income, Length: 30694, dtype: bool

In [27]:
df.isna().sum()

age               0
workclass         0
fnlwgt            0
education         0
education-num     0
marital-status    0
occupation        0
relationship      0
race              0
sex               0
capital-gain      0
capital-loss      0
hours-per-week    0
native-country    0
annual_income     0
capital_change    0
dtype: int64

In [28]:
32561-32537



24

In [29]:
df.duplicated().sum()

0

In [30]:
df.head()

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country annual_income  \
0          2174             0              40  United-States         <=50K   
1             0             0              13  United-States         <=50K   
2             0             0              40  United-States         <=50K   
3             0             0              40  United-States         <=50K   
4             0             0              40           Cuba         <=50K   

   capital_change  
0            2174  
1               0  
2               0  
3               0  
4               0

In [31]:
df.columns

Index(['age', 'workclass', 'fnlwgt', 'education', 'education-num',
       'marital-status', 'occupation', 'relationship', 'race', 'sex',
       'capital-gain', 'capital-loss', 'hours-per-week', 'native-country',
       'annual_income', 'capital_change'],
      dtype='object')

In [32]:
#df.drop(columns=['education','occupation','relationship','race','native-country','capital-gain', 'capital-loss'],axis=1,inplace=True)

In [33]:
df.columns

Index(['age', 'workclass', 'fnlwgt', 'education', 'education-num',
       'marital-status', 'occupation', 'relationship', 'race', 'sex',
       'capital-gain', 'capital-loss', 'hours-per-week', 'native-country',
       'annual_income', 'capital_change'],
      dtype='object')

In [34]:
df['annual_income'].value_counts()

<=50K    23046
>50K      7648
Name: annual_income, dtype: int64

In [35]:
#To assign a number to the annual income column
df['annual_income'] = df['annual_income'].map({'<=50K':0,'>50K':1})
df.head()

C:\Users\Avita\AppData\Local\Temp\ipykernel_10008\25930999.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['annual_income'] = df['annual_income'].map({'<=50K':0,'>50K':1})


age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country  annual_income  \
0          2174             0              40  United-States              0   
1             0             0              13  United-States              0   
2             0             0              40  United-States              0   
3             0             0              40  United-States              0   
4             0             0              40           Cuba              0   

   capital_change  
0            2174  
1               0  
2               0  
3               0  
4               0

In [36]:
df

age         workclass  fnlwgt   education  education-num  \
0       39         State-gov   77516   Bachelors             13   
1       50  Self-emp-not-inc   83311   Bachelors             13   
2       38           Private  215646     HS-grad              9   
3       53           Private  234721        11th              7   
4       28           Private  338409   Bachelors             13   
...    ...               ...     ...         ...            ...   
32556   27           Private  257302  Assoc-acdm             12   
32557   40           Private  154374     HS-grad              9   
32558   58           Private  151910     HS-grad              9   
32559   22           Private  201490     HS-grad              9   
32560   52      Self-emp-inc  287927     HS-grad              9   

           marital-status         occupation   relationship   race     sex  \
0           Never-married       Adm-clerical  Not-in-family  White    Male   
1      Married-civ-spouse    Exec-managerial        Husband  White    Male   
2                Divorced  Handlers-cleaners  Not-in-family  White    Male   
3      Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4      Married-civ-spouse     Prof-specialty           Wife  Black  Female   
...                   ...                ...            ...    ...     ...   
32556  Married-civ-spouse       Tech-support           Wife  White  Female   
32557  Married-civ-spouse  Machine-op-inspct        Husband  White    Male   
32558             Widowed       Adm-clerical      Unmarried  White  Female   
32559       Never-married       Adm-clerical      Own-child  White    Male   
32560  Married-civ-spouse    Exec-managerial           Wife  White  Female   

       capital-gain  capital-loss  hours-per-week native-country  \
0              2174             0              40  United-States   
1                 0             0              13  United-States   
2                 0             0              40  United-States   
3                 0             0              40  United-States   
4                 0             0              40           Cuba   
...             ...           ...             ...            ...   
32556             0             0              38  United-States   
32557             0             0              40  United-States   
32558             0             0              40  United-States   
32559             0             0              20  United-States   
32560         15024             0              40  United-States   

       annual_income  capital_change  
0                  0            2174  
1                  0               0  
2                  0               0  
3                  0               0  
4                  0               0  
...              ...             ...  
32556              0               0  
32557              1               0  
32558              0               0  
32559              0               0  
32560              1           15024  

[30694 rows x 16 columns]

In [37]:
#Replace the annual income col name
#df["annual_income"].replace(["Annual Income"],inplace=True)
df.rename(columns={'Annual Income':'annual_income'}, inplace = True)

C:\Users\Avita\AppData\Local\Temp\ipykernel_10008\647813675.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.rename(columns={'Annual Income':'annual_income'}, inplace = True)


In [38]:
df

age         workclass  fnlwgt   education  education-num  \
0       39         State-gov   77516   Bachelors             13   
1       50  Self-emp-not-inc   83311   Bachelors             13   
2       38           Private  215646     HS-grad              9   
3       53           Private  234721        11th              7   
4       28           Private  338409   Bachelors             13   
...    ...               ...     ...         ...            ...   
32556   27           Private  257302  Assoc-acdm             12   
32557   40           Private  154374     HS-grad              9   
32558   58           Private  151910     HS-grad              9   
32559   22           Private  201490     HS-grad              9   
32560   52      Self-emp-inc  287927     HS-grad              9   

           marital-status         occupation   relationship   race     sex  \
0           Never-married       Adm-clerical  Not-in-family  White    Male   
1      Married-civ-spouse    Exec-managerial        Husband  White    Male   
2                Divorced  Handlers-cleaners  Not-in-family  White    Male   
3      Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4      Married-civ-spouse     Prof-specialty           Wife  Black  Female   
...                   ...                ...            ...    ...     ...   
32556  Married-civ-spouse       Tech-support           Wife  White  Female   
32557  Married-civ-spouse  Machine-op-inspct        Husband  White    Male   
32558             Widowed       Adm-clerical      Unmarried  White  Female   
32559       Never-married       Adm-clerical      Own-child  White    Male   
32560  Married-civ-spouse    Exec-managerial           Wife  White  Female   

       capital-gain  capital-loss  hours-per-week native-country  \
0              2174             0              40  United-States   
1                 0             0              13  United-States   
2                 0             0              40  United-States   
3                 0             0              40  United-States   
4                 0             0              40           Cuba   
...             ...           ...             ...            ...   
32556             0             0              38  United-States   
32557             0             0              40  United-States   
32558             0             0              40  United-States   
32559             0             0              20  United-States   
32560         15024             0              40  United-States   

       annual_income  capital_change  
0                  0            2174  
1                  0               0  
2                  0               0  
3                  0               0  
4                  0               0  
...              ...             ...  
32556              0               0  
32557              1               0  
32558              0               0  
32559              0               0  
32560              1           15024  

[30694 rows x 16 columns]

In [39]:
#To combine various columns of Marital Status to married and single status
df["marital-status"] = df["marital-status"].replace(['Never-married','Divorced','Seperated','Widowed'],'Single')
df["marital-status"] = df["marital-status"].replace(['Married-civ-spouse','Married-spouse-absent','Married-AF-spouse'],'Married')
df["marital-status"] = df["marital-status"].map({"Married":1,"Single":0})


C:\Users\Avita\AppData\Local\Temp\ipykernel_10008\1288866827.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["marital-status"] = df["marital-status"].replace(['Never-married','Divorced','Seperated','Widowed'],'Single')
C:\Users\Avita\AppData\Local\Temp\ipykernel_10008\1288866827.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["marital-status"] = df["marital-status"].replace(['Married-civ-spouse','Married-spouse-absent','Married-AF-spouse'],'Married')
C:\Users\Avita\AppData\Local\Temp\ipykernel

In [40]:
df['marital-status'].value_counts()

0.0    14992
1.0    14743
Name: marital-status, dtype: int64

In [41]:
df

age         workclass  fnlwgt   education  education-num  \
0       39         State-gov   77516   Bachelors             13   
1       50  Self-emp-not-inc   83311   Bachelors             13   
2       38           Private  215646     HS-grad              9   
3       53           Private  234721        11th              7   
4       28           Private  338409   Bachelors             13   
...    ...               ...     ...         ...            ...   
32556   27           Private  257302  Assoc-acdm             12   
32557   40           Private  154374     HS-grad              9   
32558   58           Private  151910     HS-grad              9   
32559   22           Private  201490     HS-grad              9   
32560   52      Self-emp-inc  287927     HS-grad              9   

       marital-status         occupation   relationship   race     sex  \
0                 0.0       Adm-clerical  Not-in-family  White    Male   
1                 1.0    Exec-managerial        Husband  White    Male   
2                 0.0  Handlers-cleaners  Not-in-family  White    Male   
3                 1.0  Handlers-cleaners        Husband  Black    Male   
4                 1.0     Prof-specialty           Wife  Black  Female   
...               ...                ...            ...    ...     ...   
32556             1.0       Tech-support           Wife  White  Female   
32557             1.0  Machine-op-inspct        Husband  White    Male   
32558             0.0       Adm-clerical      Unmarried  White  Female   
32559             0.0       Adm-clerical      Own-child  White    Male   
32560             1.0    Exec-managerial           Wife  White  Female   

       capital-gain  capital-loss  hours-per-week native-country  \
0              2174             0              40  United-States   
1                 0             0              13  United-States   
2                 0             0              40  United-States   
3                 0             0              40  United-States   
4                 0             0              40           Cuba   
...             ...           ...             ...            ...   
32556             0             0              38  United-States   
32557             0             0              40  United-States   
32558             0             0              40  United-States   
32559             0             0              20  United-States   
32560         15024             0              40  United-States   

       annual_income  capital_change  
0                  0            2174  
1                  0               0  
2                  0               0  
3                  0               0  
4                  0               0  
...              ...             ...  
32556              0               0  
32557              1               0  
32558              0               0  
32559              0               0  
32560              1           15024  

[30694 rows x 16 columns]

# Test Train Data

In [42]:
x=df.iloc[:,1:-1]

In [43]:
x

workclass  fnlwgt   education  education-num  marital-status  \
0             State-gov   77516   Bachelors             13             0.0   
1      Self-emp-not-inc   83311   Bachelors             13             1.0   
2               Private  215646     HS-grad              9             0.0   
3               Private  234721        11th              7             1.0   
4               Private  338409   Bachelors             13             1.0   
...                 ...     ...         ...            ...             ...   
32556           Private  257302  Assoc-acdm             12             1.0   
32557           Private  154374     HS-grad              9             1.0   
32558           Private  151910     HS-grad              9             0.0   
32559           Private  201490     HS-grad              9             0.0   
32560      Self-emp-inc  287927     HS-grad              9             1.0   

              occupation   relationship   race     sex  capital-gain  \
0           Adm-clerical  Not-in-family  White    Male          2174   
1        Exec-managerial        Husband  White    Male             0   
2      Handlers-cleaners  Not-in-family  White    Male             0   
3      Handlers-cleaners        Husband  Black    Male             0   
4         Prof-specialty           Wife  Black  Female             0   
...                  ...            ...    ...     ...           ...   
32556       Tech-support           Wife  White  Female             0   
32557  Machine-op-inspct        Husband  White    Male             0   
32558       Adm-clerical      Unmarried  White  Female             0   
32559       Adm-clerical      Own-child  White    Male             0   
32560    Exec-managerial           Wife  White  Female         15024   

       capital-loss  hours-per-week native-country  annual_income  
0                 0              40  United-States              0  
1                 0              13  United-States              0  
2                 0              40  United-States              0  
3                 0              40  United-States              0  
4                 0              40           Cuba              0  
...             ...             ...            ...            ...  
32556             0              38  United-States              0  
32557             0              40  United-States              1  
32558             0              40  United-States              0  
32559             0              20  United-States              0  
32560             0              40  United-States              1  

[30694 rows x 14 columns]

In [44]:
from sklearn.model_selection import train_test_split

In [45]:
x_train,y_train,x_test,y_test= train_test_split(x,y,test_size=.7,random_state=0)

NameError: name 'y' is not defined

# 2 Question
#MyAnswer - 283

# 3 Question
#My Answer - 0

# 5 Question
#my Answer - 2554

# 6 Question
#17679

# Need help with these  15,16,17 & 18 questions